### DISCUSSION TITLE
Using GitHub Secrets without using GitHub Actions

### DISCUSSION THREAD

**Discussion by shayanorellana** [community/community#26659] [Actions]
<p>I was wondering if it’s at all possible to use GitHub secrets in python code located in the repository without initializing the secrets in the env section of the GitHub Actions yaml file. I have not set up my GitHub Actions workflows yet for the scripts, was just wondering if it was possible to use the GitHub secrets strictly in the python code only to mask data-sensitive information. Thanks!</p>

**Comment by airtower-luna** [comment]
<p>Simple answer: No, it’d defeat the point if the secrets were accessible outside workflows.</p>
<aside class="quote no-group" data-username="shayanorellana" data-post="1" data-topic="251523">
<div class="title">
<div class="quote-controls"></div>
 shayanorellana:</div>
<blockquote>
<p>was just wondering if it was possible to use the GitHub secrets strictly in the python code only to mask data-sensitive information</p>
</blockquote>
</aside>
<p>What exactly are you trying to achieve?</p>

**Comment by shayanorellana** [comment]
<p>I have some sensitive data including usernames and passwords in the repository and was wondering if I could use Secrets to mask these in the script. But your answer makes sense.</p>
<p>I was following this article: <a href="https://medium.datadriveninvestor.com/accessing-github-secrets-in-python-d3e758d8089b" class="inline-onebox" rel="noopener nofollow ugc">Accessing GitHub secrets in Python | by Dipam Vasani | DataDrivenInvestor</a></p>
<p>But I have to setup a yaml workflow for the secrets initialized in the yaml ENV and then use os.environ[‘SECRETS_KEY’] in the script for this to work. Is this correct?</p>

**Comment by airtower-luna** [comment]
<p>What exactly do you mean by “masking”? All the masking the Actions runner does is in the logs, if you provide secrets to your script the script has them in plaintext. To achieve the same thing in your script you can just not include potentially sensitive values in its output.</p>

**Comment by shayanorellana** [comment]
<p>For example I have a script that requires the mongoDB login credentials and want to mask or hide this information to the script. I was looking for alternatives such as Secrets to be able to store this information and use os.environ[‘MONGODB_PASSWORD’] in the script to do so.</p>
<p>mongodb_pass = os.environ[‘MONGODB_PASSWORD’]</p>
<p>ex: MongoClient(username: mongodb_pass) instead of having the password in plain sight in the script. But since this information is not in the workflow (because not using Actions) then it wouldn’t work, correct?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="shayanorellana" data-post="5" data-topic="251523">
<div class="title">
<div class="quote-controls"></div>
 shayanorellana:</div>
<blockquote>
<p>ex: MongoClient(username: mongodb_pass) instead of having the password in plain sight in the script.</p>
</blockquote>
</aside>
<p>Not having the credentials directly in the script is good practice, yes. That has nothing to do with masking, though.</p>
<p>You can use environment variable just like in the Actions workflow, you’ll just have to set them when you run the script. Just be aware of what else you run in the same environment and of shell history (if you use Bash, <code>HISTCONTROL</code> is your friend).</p>
<p>An alternative is to use a config file, just read the values from another file instead of coding them into the script. Python has <a href="https://docs.python.org/3/library/json.html">built-in parsers for JSON</a> and <a href="https://docs.python.org/3/library/configparser.html">INI-style files</a>. I like YAML, but that requires <a href="https://pyyaml.org/wiki/PyYAMLDocumentation">installing PyYAML</a>. Obviously don’t commit that file, I recommend adding it to <code>.gitignore</code>. :wink:</p>
<p>You can even support both in your script: Read environment variables or config file depending on command line options or simply what’s available. :smiley_cat:</p>

**Comment by shayanorellana** [comment]
<p>Thanks for your help <a class="mention" href="/airtower-luna">@airtower-luna</a> !</p>
